deps/v8/src/builtins.cc

```diff
@@ -481,9 +481,13 @@ void Builtins::Generate_ObjectHasOwnProperty(
   assembler->Bind(&keyisindex);
   {
     Label if_objectissimple(assembler);
-    assembler->Branch(assembler->Int32LessThanOrEqual(
-                          instance_type, assembler->Int32Constant(
+    assembler->Branch(assembler->Word32And(
+                          assembler->Int32LessThanOrEqual(
+                              instance_type, assembler->Int32Constant(
                                              LAST_CUSTOM_ELEMENTS_RECEIVER)),
+                          assembler->Word32NotEqual(
+                              instance_type, assembler->Int32Constant(
+                                             JS_OBJECT_TYPE))),
```

Comment:
I think this is not right. JS_SPECIAL_API_OBJECT_TYPE used to be <= LAST_CUSTOM_ELEMENTS_RECEIVER before your change, but is > after your change. JS_OBJECT_TYPE is invariant here.

I think the expression you want is:

OR (
     LESS_EQUAL (instance_type, LAST_CUSTOM_ELEMENTS_RECEIVER
     EQUAL(instance_type, JS_SPECIAL_API_OBJECT_TYPE))